### DISCUSSION TITLE
Subtract expenses from project budget

### DISCUSSION THREAD

**Discussion by mcfly3001** [kimai/Paid-plugins#69] [Questions & Answers] [answered]
In my projects I often have to send monthly invoices with a fixed amount. This is independant from how many hours were spent during the month. For example, the customer orders 1.200h for this year with the total costs of 120.000€. The invoicing will be done on a monthly basis, i.e. each month 10.000€.
Hence, the progress of the worked hours and the invoiced budget can differ throughout the year. It's only important that by end of the project, the ordered hours were delivered.

I wanted to do it as follows in Kimai:

- Add activities for Development, Meetings, Others as billable activities but with an hourly rate of 0€. This way, the hourly progress can be tracked.
- Each month, add an expense entry worth 10.000€ for our service in the last month. I was expecting to see an increase of the used budget as soon as I would mark the expense as paid. This way, I could easily update the customer on how much budget was already spent compared to the actual work hours we delivered.

My second assumption though does not work, because the expenses are not subtracted from the overall budget. What is the reason for this? Or is there a better way to do that?

Thanks for any hints!

**Comment by kevinpapst** [comment, marked as answer]
Thanks for re-posting! Better to collect all discussion in one place.

There is a configuration for Expenses, that needs to be activated:
<img width="478" alt="Bildschirmfoto 2023-05-10 um 13 13 41" src="https://github.com/kevinpapst/kimai-plugins/assets/533162/890e3505-79fc-419c-b49c-2979fb21d06f">

Pleas retry and check, if it does what you want.

**Reply by mcfly3001** [reply]
Thanks for pointing me to the configuration. Did not check for configuration options of that plugin. Works now as expected, thanks!

**Reply by mcfly3001** [reply]
Hello again,

I did some testing today and actually think that I would not like to set the hourly rates of an activity to 0€. Ideally I would be able to set the hourly rate (e.g. 100€), internal rate (e.g. 50€) per user as done before, such that I can track the revenue.
Invoicing though should **only** be based on the expenses. Would this be possible as well? Currently I am struggling to configure an invoicing template which only shows the expenses and explicitely does not add the billable hours. I need to label the hours as billable, otherwise the hours would not get tracked in the project progress overview.

Probably I will have to create my own invoicing template for this, right?

Thanks for your feedback!

**Reply by mcfly3001** [reply]
One possible workaround I think I just found is to do as follows:

- Add the project activities as usual and enter the hourly rates as usual
- Add an additional activity, just for "invoicing" where on a monthly basis an expenses entry will be added
- For the invoicing, use the activity filter and only select the "invoicing" activity to create the invoice
- (optionally): also mark the other activities as exported

In this case, in the settings the expenses should not be added to the consumed budget of course. Otherwise, at the end of the year, the spent budget would be double of the actual budget.

Any better ideas?
